### DISCUSSION TITLE
WSL2 tab auto complete stuck/slow

### DISCUSSION THREAD

**Discussion by talhag3** [microsoft/WSL#5929] [General]
I have ubuntu 20 install in WSL2 . 
when i type 
#   service mys(press tab , it stuck and response after 10sec every time

**Comment by therealkenc** [comment]
Onomatopellan's suggestion is a good start, because "slow anything" usually boils down to #4197. Also make sure you haven't mucked `$LD_LIBRARY_PATH` like #5696.

This all said, when I did an `strace(1)`, the `bash_completion` script is behaving oddly for the `service` rule. It loops poking at `90-systemd.preset1` more than 300 times.

![image](https://user-images.githubusercontent.com/7049700/93815656-31e2ac00-fc0b-11ea-84b3-2831dc10355c.png)

Here is the log if someone feels motivated to look into why. It isn't going to to be WSL actionable (it is doing that for _reasons_) but might (big might) be because we don't run `systemd`. On this run it starts poking around `systemd` at around `13:08:49` and takes all the way to `13:09:25` to give up on whatever it is trying to do. Those wall clock times are exaggerated by the time to trace and log.

[bash.strace.zip](https://github.com/microsoft/WSL/files/5257927/bash.strace.zip)
